src/components/Characters.js

```diff
@@ -29,78 +30,185 @@ export default function Characters(props) {
 
   const characters = [...swrCharacters.data.persons, ...swrNpcs.data.persons];
 
-  function getRowIndex(cell, row, rowIndex) {
-    return (page - 1) * sizePerPage + rowIndex + 1;
-  }
-
   const selectOptions = {
     true: 'Character',
     false: 'NPC'
   };
 
   const selectPlanet = toSelectOptions(characters, 'home_planet');
   const selectDynasty = toSelectOptions(characters, 'dynasty');
+  const selectParty = toSelectOptions(characters, 'political_party');
+  const selectReligion = toSelectOptions(characters, 'religion');
+  const selectShift = toSelectOptions(characters, 'shift');
   const selectStatus = toSelectOptions(characters, 'status');
   const selectShip = toSelectOptions(characters.map((c) => c.ship), 'name');
 
+  const { ToggleList } = ColumnToggle;
+
   const columns = [{
     dataField: 'id',
     text: 'Id',
     sort: true,
     headerStyle: () => {
       return { width: '60px', textAlign: 'center' };
     },
-    align: 'center'
+    align: 'center',
+    sortCaret: (order, column) => {
+      if (!order) return (<span>&nbsp;<font color="gray">↑↓</font></span>);
+      else if (order === 'asc') return (<span>&nbsp;↑<font color="gray">↓</font></span>);
+      else if (order === 'desc') return (<span>&nbsp;<font color="gray">↑</font>↓</span>);
+      return null;
+    }
```

Comment:
This repeats so many times that it's better to just define it as a function and refer to it.

```js
// Define this above
const sortCaret = (order, column) => {
  if (!order) return (<span>&nbsp;<font color="gray">↑↓</font></span>);
  else if (order === 'asc') return (<span>&nbsp;↑<font color="gray">↓</font></span>);
  else if (order === 'desc') return (<span>&nbsp;<font color="gray">↑</font>↓</span>);
  return null;
}
```

And then refer to it here by just using a shorthand, like
```js
const columns = [{
  // ...stuff
  align: 'center',
  sortCaret,
  // ...more stuff
}];
```